server/src/actions/deleter.ts

```diff
@@ -0,0 +1,25 @@
+import { messagesCollection } from '../utilities/firebaseInit'
+const cron = require('node-cron')
+
+//Schedule tasks to be run on the server
+export const scheduleCron = () => {
+    cron.schedule('*/30 * * * * *', function() {
+        console.log('Deleting old messages every 30s.')
+    
+        //Deleter action, takes in a unix timestamp and deletes
+        //everything older than that
+        const expiryTime = Number(process.env.message_duration) //Set to 1 minute for testing purposes
+    
+        const q = messagesCollection.orderByChild('timeSent').endAt(Date.now() - expiryTime)
+        
+        q.on('value', (querySnapshot)=> {
+            querySnapshot.forEach(async (doc) => {
+                //Delete the doc here
+                console.log(doc.id)
+                await doc.ref.delete()
```

Comment:
For now, could you actually acknowledge each doc you delete and console.log() the msgId beign deleted?

Reply:
It already `console.log` the msgId